client/ayon_maya/plugins/load/load_layout.py

```diff
@@ -0,0 +1,233 @@
+        if not version_ids:
+            return output
+
+        project_name = get_current_project_name()
+        repre_entities = ayon_api.get_representations(
+            project_name,
+            representation_names={"fbx", "abc"},
+            version_ids=version_ids,
+            fields={"id", "versionId", "name"}
+        )
+        for repre_entity in repre_entities:
+            version_id = repre_entity["versionId"]
+            output[version_id].append(repre_entity)
+        return output
+
+    @staticmethod
+    def _get_loader(loaders, product_type):
+        name = ""
+        if product_type in {
+            "rig", "model", "camera",
+            "animation", "staticMesh",
+            "skeletalMesh"}:
+                name = "ReferenceLoader"
+
+        if name == "":
+            return None
+
+        for loader in loaders:
+            if loader.__name__ == name:
+                return loader
+
+        return None
+
+    def get_asset(self, instance_name):
+        container = [
+            con for con in cmds.ls(f"{instance_name}*")
+            if con.endswith("_CON")
+        ][0]
+        namespace = cmds.getAttr(f"{container}.namespace")
+        asset = [asset for asset in cmds.ls(f"{namespace}:*", assemblies=True)][0]
+        return asset
+
+    def _process_element(self, filepath, options, loaded_options=None):
+
+        with open(filepath, "r") as fp:
+            data = json.load(fp)
+
+        all_loaders = discover_loader_plugins()
+
+        if loaded_options is None:
+            loaded_options = []
+
+        # get the list of representations by using version id
+        repre_entities_by_version_id = self._get_repre_entities_by_version_id(
+            data
+        )
+        for element in data:
+            repre_id = None
+            repr_format = None
+            version_id = element.get("version")
+            if version_id:
+                repre_entities = repre_entities_by_version_id[version_id]
+                if not repre_entities:
+                    self.log.error(
+                        f"No valid representation found for version"
+                        f" {version_id}")
+                    continue
+                extension = element.get("extension")
+                # always use the first representation to load
+                repre_entity = next((repre_entity for repre_entity in repre_entities
+                                    if repre_entity["name"] == extension), None)
+                repre_id = repre_entity["id"]
+                repr_format = repre_entity["name"]
+
+            # If reference is None, this element is skipped, as it cannot be
+            # imported in Maya
+            if not repre_id:
+                continue
+
+            instance_name = element.get('instance_name')
+            containers = [
+                con for con in cmds.ls(f"{instance_name}*")
+                if con.endswith("_CON")
+            ]
```

Comment:
I'm honestly quite confused by this logic - what's this "lookup" by name? Why?

Reply:
It's looking up the container of the loaded assets. If the loaded asset is there, it won't be loaded again.

Reply:
But - how would that 'container' magically match exactly the name from the JSON data?